R/task.R

```diff
@@ -546,10 +563,11 @@ final_status_to_logical <- function(status) {
 ##'   `FALSE` otherwise.
 ##'
 ##' @export
-task_wait <- function(id, timeout = Inf, poll = 1, progress = NULL,
-                      root = NULL) {
+task_wait <- function(id, follow = TRUE, timeout = Inf, poll = 1,
+                      progress = NULL, root = NULL) {
   root <- hipercow_root(root)
-  status <- task_status(id, root = root)
+  id <- follow_retry_map(id, root)
```

Comment:
I am getting confused, but I think this one should have an if (follow) round it too - suppose...

job id1 fails.
the retry, job id2, runs for a long time, surprisingly, so we wait for it.

In that case... 
task_wait(id2, follow = FALSE) should wait.
task_wait(id2, follow = TRUE) should wait.
task_wait(id1, follow = TRUE) should wait, because we realise id2 is the new id1
task_wait(id1, follow = FALSE) should probably not wait, but I think above it will.

But this is quite hard to test/replicate!

Reply:
I suppose there's a question on why one would particularly want to do this forcing follow = FALSE